web_monitoring/diffing_server.py

```diff
@@ -367,6 +368,13 @@ def _extract_encoding(headers, content):
 
 
 def _decode_body(response, name, raise_if_binary=True):
+    """Check if the response is binary. If not, continue and detect encoding.
+    Try to define if a significantly large part of the body was undecodable
+    before returning decoded body.
+    """
+    if is_not_html(response.body, response.headers):
```

Comment:
I don’t think we should be checking for whether something is HTML here. What if we had a differ for RSS or XML or CSV documents in the future? There are plenty of valid text types that aren’t HTML. This should just be checking for whether what we got is not decodable text. We talked about this on Slack: https://archivers.slack.com/archives/C47DJK8H0/p1559529058016900

I’m also a little worried about not supporting the `content_type_options` argument at this point (its behavior is meant to be analogous to the `X-Content-Type-Options` response header). We felt it was important to have in case we ran into situations were sniffing was likely to produce a false positive. A text document *could* always start with `GIF8` and it seemed important to provide a way to handle that. Do you not think that’s important here?